src/frontend/classic/utils/storage.ts

```diff
@@ -0,0 +1,46 @@
+export type SettingValue = string | number | boolean;
+export type SettingExtras = ((extrasArg?: SettingValue) => void) | undefined
+export type SettingValidator = ((value: SettingValue) => boolean) | undefined
+
+export const onSettingChange = (
```

Comment:
Use svelte stores. Probably one dedicated to each kind ? ("General","UI").

```typescript
import { writable } from "svelte/store";

const storedValue= localStorage.getItem("thisvalue");
export const readthisvalue = writable(storedValue);
readthisvalue .subscribe(value => {
    localStorage.setItem("thisvalue", value === 'mooo' ? 'mooo' : 'meh');
});
```

Reply:
Ok, I wasn't sure if we wanted to use svelte store, so I pick up the easiest option ^^